pkg/util/net.go

```diff
@@ -52,3 +53,21 @@ func RetryInsecureHttpsGet(url string) int {
 
 	return status
 }
+
+func RetryTCPConnection(host string, port string) bool {
+	status := true
+	err := wait.Poll(5*time.Second, 120*time.Second, func() (bool, error) {
+		timeout := 30 * time.Second
+		_, err := tcpnet.DialTimeout("tcp", tcpnet.JoinHostPort(host, port), timeout)
+
+		if err == nil {
+			status = false
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil && err == wait.ErrWaitTimeout {
+		logrus.Warningf("Endpoint is not returning any status code")
```

Comment:
if timeout, this should return false to tell the caller the connection is not ready